### ISSUE TITLE
Source code links not working for properties

### ISSUE THREAD

**Issue by pcuenca** [fastai/fastai_dev#187]
See the last few cells in notebook `05_data_core`, where we attempt to `show_doc` `DataBunch` properties  `train_dl`, `valid_dl`, `train_ds` and `valid_ds`. The documentation summary works fine, but the source code links come up empty.

Those properties are added via `add_props`, which in turn creates them as `property` instances. The function [`get_name`](https://github.com/fastai/fastai_dev/blob/master/dev/local/notebook/export.py#L260), defined in `91_notebook_export`, is unable to derive a symbolic name for the desired property.

I haven't found a way to introspect the property name to be able to find the notebook where the symbol is defined. As an alternative, we could add something like the following
to obtain the name of the **object** the property has been added to:

```Python
# export
def get_name(obj):
    "Get the name of `obj`"
    if hasattr(obj, '__name__'):       return obj.__name__
    elif getattr(obj, '_name', False): return obj._name
    elif hasattr(obj,'__origin__'):    return str(obj.__origin__).split('.')[-1] #for types
    elif type(obj) == property:        return obj.fget.func.__qualname__.split('.')[0]
    else:                              return str(obj).split('.')[-1]
```

If we were to invoke `get_name(DataBunch.train_dl)`, `get_name` would return `DataBunch`. This is not semantically correct, but it would suffice to make the rest of the infrastructure point to the source code for `DataBunch`, which is the same result we get for other fields.

Is this something worth pursuing? If so, is there a way to introspect the property name instead of the parent object's name, or should we be satisfied with the latter?

(This is similar to the issue addressed in https://github.com/fastai/fastai_dev/pull/170, but the root cause is different).

**Comment by sgugger** [comment]
I didn't know of this way to get the owner of a property, thanks for pointing that out. `get_name` should return the full name (`DataBunch.train_dl` for instance) since then the function that looks for the corresponding notebook will try `DataBunch.train_dl` then `DataBunch`, and this way we get the real name of the property. Would you mind making a PR for that?

Not that the source links in `show_doc` pointing to notebooks is temporary as there is a bug with the links generated to the source code in github during the execution of the show_doc cells, so this will eventually fix the function `nb_source_link` for the long run.

**Comment by sgugger** [comment]
Update, source links are now all pointing to Github since [this commit](https://github.com/fastai/fastai_dev/commit/671199cc75f84fb3e5225bc14b6e55c7ff0bc681).

**Comment by pcuenca** [comment]
> Update, source links are now all pointing to Github since [this commit](https://github.com/fastai/fastai_dev/commit/671199cc75f84fb3e5225bc14b6e55c7ff0bc681).

Understood, I'll submit a PR. Thanks!

**Comment by sgugger** [comment]
Closed by #201
